app/routes/datasets/dataset.py

```diff
@@ -55,7 +55,11 @@ async def create_dataset(
     is_authorized: bool = Depends(is_admin),
     response: Response,
 ) -> DatasetResponse:
-    """Create or update a dataset."""
+    """Create a dataset. a “dataset” is largely a metadata concept: it represents
+    a data product that may have multiple versions or file formats over time.
+
+    This operation requires a `MANAGER` or an `ADMIN` user role.
```

Comment:
Maybe add an extra sentence here:

The user that creates a dataset using this operation becomes the `owner` of the dataset, which provides the user with the privileges to do further write operations on the dataset, including creating and modifying versions and assets.